app/src/main/kotlin/com/wire/android/ui/home/conversations/details/GroupConversationDetailsViewModel.kt

```diff
@@ -79,51 +81,60 @@ class GroupConversationDetailsViewModel @Inject constructor(
-                            combine(
-                                conversationDetailsFlow,
-                                isSelfAdminFlow,
-                                getSelfTeam()
-                            ) { conversationDetails, isSelfAnAdmin, selfTeam ->
-                                Triple(
-                                    conversationDetails,
-                                    isSelfAnAdmin,
-                                    selfTeam
-                                )
-                            }.collect { (conversationDetails, isSelfAnAdmin, selfTeam) ->
-                                with(conversationDetails) {
-                                    val isSelfInOwnerTeam =
-                                        selfTeam?.id != null && selfTeam.id == conversationDetails.conversation.teamId?.value
+        viewModelScope.launch(dispatcher.io()) {
+
+            val groupDetailsFlow = observeConversationDetails(conversationId)
+                .filterIsInstance<ObserveConversationDetailsUseCase.Result.Success>()
+                .map { it.conversationDetails }
+                .filterIsInstance<ConversationDetails.Group>()
+                .distinctUntilChanged()
+                .shareIn(this, SharingStarted.WhileSubscribed(), 1)
+
+            val isSelfAdminFlow = observeConversationMembers(conversationId)
+                .map { it.isSelfAnAdmin }
+                .distinctUntilChanged()
 
-                                    if (this is ConversationDetails.Group) {
-                                        updateState(
-                                            groupOptionsState.copy(
-                                                groupName = conversation.name.orEmpty(),
-                                                protocolInfo = conversation.protocol,
-                                                isUpdatingAllowed = isSelfAnAdmin,
-                                                areAccessOptionsAvailable = conversation.isTeamGroup(),
-                                                isGuestAllowed = (conversation.isGuestAllowed() || conversation.isNonTeamMemberAllowed()),
-                                                isServicesAllowed = conversation.isServicesAllowed(),
-                                                isUpdatingGuestAllowed = isSelfAnAdmin && isSelfInOwnerTeam,
-                                                isAbleToRemoveGroup = selfUser.teamId != null
-                                                        && conversation.creatorId.value == selfUser.id.value
-                                            )
-                                        )
-                                    }
-                                }
-                            }
+            launch {
+                groupDetailsFlow
+                    .collect { groupDetails ->
+                        with(groupDetails) {
+                            updateState(
```

Comment:
This would be executed on the main dispatcher 🤔  ?  I think we have to be sure to avoid some runtime crashes, while trying to update state from `io`

Reply:
good point! 
i changed `groupOptionsState` to `StateFlow`, and now we can update it from any dispatcher